pkg/gui/controllers/branches_controller.go

```diff
@@ -426,7 +426,7 @@ func (self *BranchesController) createNewBranchWithName(newBranchName string) er
 	}
 
 	self.context().SetSelectedLineIdx(0)
-	return self.c.Refresh(types.RefreshOptions{Mode: types.ASYNC})
+	return self.c.Refresh(types.RefreshOptions{Mode: types.ASYNC, KeepBranchSelectionIndex: true})
```

Comment:
This is a very specific field we're introducing, but we're going to need one of these for any sortable list context. Thoughts on just calling it `KeepSelectionIndex` and letting it act on the current static context? I don't imagine we would ever need to support retaining the selection index on non-focused context.

Reply:
> I don't imagine we would ever need to support retaining the selection index on non-focused context.

I think it's a good idea to retain the branch selection even when the branches list is not focused, for example when quickly going to the files panel to press "f" and then coming back to the branches panel. So I'd vote for keeping this field specific to branches.

Case in point: I have a branch (will make a PR soon 😄) that retains the selection after squashing fixup commits (right now we keep the same index). At first I thought that would only be a minor cosmetic thing, but I was actually surprised how much this contributes to the perception of robustness or stability, whatever you want to call it.

Reply:
> I think it's a good idea to retain the branch selection even when the branches list is not focused, for example when quickly going to the files panel to press "f" and then coming back to the branches panel. So I'd vote for keeping this field specific to branches.

Wait, I thought that was default behaviour anyway, and that the flag only existed to do the opposite i.e. to retain the selection _index_ but not necessarily the selected _item_. I suppose you could still make a similar argument though for that case. I do feel uncomfortable that we'd need to add a new flag to this refresh args struct whenever we want to support sorting in a new view, but I can't think of a better solution